### ISSUE TITLE
Nest the current face shield design to optimize cut time and yield

### ISSUE THREAD

**Issue by MatthewCroughan** [DoESLiverpool/covid19#66]
I'm currently running a job to generate the best possible placement of the odd shapes we have to deal with. The sheet in this case is 1000x600, just to be safe since I'm not sure what the dimensions are of the available materials are, and we don't want to go right up to the edge as that would require more precision and better sheet placement.

Additionally, this nesting program will place each item as close together as needed to get the same part from only a single line cut. So, for example, if these lines overlap then instead of performing two cuts, it will only perform one, the result is that both pieces benefit from this overlap. Those lines are shown in blue in the images below.

My opinion is that we should take a single sheet and cut one specific part of this 3 part design out, as mixing and matching them seems to be less efficient, then take the next sheet and produce the next part of the design.

![image](https://user-images.githubusercontent.com/26458780/79049266-858fa900-7c1a-11ea-9e23-35af9f09d75f.png)

![image](https://user-images.githubusercontent.com/26458780/79049140-8e33af80-7c19-11ea-83f1-f07b2508cea5.png)

**Comment by ajlennon** [comment]
Nice idea. I was wondering about the "Adam Smith" approach myself. 

One thing I would suggest is that it is a complete b** turning a raw DXF into a Laser Cut project in terms of setting cutting orders.

So if you're going to create a new DXF to try can you also provide a new Laser Cut .ecp project.

**Comment by MatthewCroughan** [comment]
There's some program that Abdul from DoES showed me a while ago that allows you to create such a thing. Laser Cut 5.0 allows you to import more than just DXFs, apparently you can encode that stuff in other formats and it'll pass through. I'll look into this, though I need to get some sleep first.

**Comment by goatchurchprime** [comment]
There's a cost to packing things in too cleverly on what is fairly cheap material.  If it was gold leaf we'd need to do this.  But given what it is, a more regular and predictable cut is useful.  

Maybe it would be better to spread things out a little bit so that we didn't get so many large smoke marks across all the parts.

**Comment by MatthewCroughan** [comment]
When I was watching it cut, the smoke marks appear because the laser cuts over a line that has already been cut, this nesting approach with the parts being close enough together to only perform one cut would solve that problem, at least I would have thought. @goatchurchprime

**Comment by ajlennon** [comment]
>Maybe it would be better to spread things out a little bit so that we didn't get so many large smoke marks across all the parts.

Could be. All I can say from my time in front of the cutter is it we optimise for speed and go as fast as poss. then the smoke marks are significantly lessened.

Also wherever we can share edges of shapes we obviously get in cutting time...

**Comment by huffeec** [comment]
The white sheet on the pallet is 720x1020 
Can you update this page if you change the layouts.
https://github.com/DoESLiverpool/covid19/blob/master/visor-designs/laser-cut/README.md

Looks like you're using the kitronik design there? Is than an example or have we switched? It takes visors with different hole patterns.

**Comment by ajlennon** [comment]
>When I was watching it cut, the smoke marks appear because the laser cuts over a line that has already been cut, this nesting approach with the parts being close enough together to only perform one cut would solve that problem, at least I would have thought. @goatchurchprime

Yes I'd agree with that @MatthewCroughan and I've reworked the project now so it doesn't go over lines twice.

**Comment by MatthewCroughan** [comment]
@huffeec I'm glad you've mentioned that. The https://github.com/DoESLiverpool/covid19/blob/master/visor-designs/laser-cut/production.md states that we should be using this design. It is what I cut last night. 

What design should we be cutting? I'm heading in every day from 11PM to do this cutting, so I need to figure out what design we're all supposed to be using.

**Comment by goatchurchprime** [comment]
What's the point of these tiny little rads here in the line?  They are the source of unnecessary burn marks as well.

![image](https://user-images.githubusercontent.com/677254/79049485-dd7adf80-7c1b-11ea-98fa-7ef78fe24bb6.png)

**Comment by MatthewCroughan** [comment]
@goatchurchprime I'm thinking now that this is the wrong design to be using, so I'm going to try the other.

**Comment by ajlennon** [comment]
>Looks like you're using the kitronik design there? Is than an example or have we switched? It takes visors with different hole patterns.

Now I'm confused - I thought we were using your design at present @huffeec ? You put one together with 20 pieces on for the new stock size?

**Comment by MatthewCroughan** [comment]
@huffeec I need that design as an SVG in order to nest it, a DXF will not do because it is now split into segments that appear as entirely separate objects in the nesting program.

**Comment by huffeec** [comment]
> I need to figure out what design we're all supposed to be using.

AFAIK it' still the file listed at the top of the readme This is the one Adrian has approved by dr's

**Comment by MatthewCroughan** [comment]
Alright, seems things are a bit fragmented at the moment, thanks for clarifying. I will be cutting that tonight instead when I'm in DoES.

**Comment by ajlennon** [comment]
@MatthewCroughan - I've spent most of today getting the design we are cutting right. Let's not change this until we understand what's going on here. Nobody here has flagged we're cutting the wrong design and we have been cutting a lot.

**Comment by huffeec** [comment]
Neither are really my design. This is Adrian's original laser cut design with the kitronik strap added in because we couldn't source elastic. There never was an SVG it's put together from two DXF files. 

I put a couple of single line sheets together for the two material sized I knew we'd ordered. If it can be optimised that's great. Main thing is we crank these out and get them on heads now we're not short of material.

**Comment by MatthewCroughan** [comment]
Ah, okay. There is nothing to worry about. This is a mental mixup. I cut the correct one at DoES last night. I haven't familiarized myself with the laser cut designs, so I visually thought this one was similar and thought it was the same when looking for an SVG of it, since an SVG is needed to nest in this circumstance.

I just ended up with this design in my search for an svg and didn't realise they were different.

I am now well aware of the differences between the two! Thanks.

**Comment by huffeec** [comment]
No worries. I think one time saving is to contract the version info. Something like v1 hybrid four holes. At the moment. Could easily be Hv1-4 or similar. And etching rather than cutting reduces the amount of cleaning.

**Comment by plastictactics** [comment]
I have been getting pretty consistent cuts with shared lines on my machine. It's overly complicated but I haven't had time to clean up yet. It has three tabs along the length of the pieces to hold things in place until the last cut, where they get snipped out like popping the parts out of an airfix kit.

![hybridv1-tabs-before-cut](https://user-images.githubusercontent.com/49289841/79050460-19b13e80-7c22-11ea-9223-cbcbc35f49d3.jpeg)

I've uploaded it over here in case it's of interest or use to someone with a smaller bed, but can also be used for Gerald. File size is huge so have zipped it. But it is in DXF and I did have a hard time getting it into Corel Draw as Alex suggested - will try and get the .crd files uploaded too just in case someone is using that and it imports more cleanly.

https://github.com/DoESLiverpool/covid19/blob/master/visor-designs/laser-cut/hybridV1x20_1020x720Sheet_split-into-smaller-panels_tabs.dxf.zip
